### PR TITLE
Migrate cloud gateway to mdx

### PR DESCRIPTION
**Pull request by dengyaxi**

### CODE REVIEW

File: shared/cloud-gateway/overview/_product-overview.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+<PlatformWrapper platform="linux">
+Deployed on your server, the Agora Cloud Gateway communicates with clients integrated with the [Agora RTC SDK](../reference/glossary#agora-video-sdk). You can use the Cloud Gateway to transmit video and audio streams between the server and the client through [SD-RTN™](../reference/glossary#agora-sd-rtn).
+
+The following diagram shows typical application scenarios of the Cloud Gateway, including playing local media files, pushing streams to CDN, and communicating with enterprise call centers. The Cloud Gateway implements media stream transmission by the following means:
+
+- Converting incoming media streams from SD&#8209;RTN™ to a specified format.
+- Encoding and sending local media streams to SD&#8209;RTN™.
+
+![1658370187632](https://web-cdn.agora.io/docs-files/1658370187632)
```
Comment (yamasite) [inline shared/cloud-gateway/overview/_product-overview.mdx:11]: Need to update the architecture diagram.

File: shared/cloud-gateway/overview/_product-overview.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Advantages
+
+The Gateway Server SDK has the following advantages:
+
+| Feature  | Description                                                         |
+| :----- | :----------------------------------------------------------- |
+| Reliability   | Supports cluster deployment, dynamic scalability, and high availability.                         |
+| Smoothness  | Low-latency media transmission from the server to the client based on the packet loss reduction feature of SD-RTN™.  |
+| High concurrency | For a server with an 8-core, 2.5 GHz CPU and 16 GB RAM, you can simultaneously send video streams (320 × 240, 15 fps) to 1,000 channels.|
```
Comment (yamasite) [inline shared/cloud-gateway/overview/_product-overview.mdx:25]: Need to update content. See: https://docs.agora.io/en/server_gateway/server_gateway_overview?platform=Linux

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+
+Complete the following steps before sending and receiving media streams.
+
+### Step 1: Initialize the SDK
+
+You need to call `createAgoraService` and `initialize` to create and initialize an `AgoraService` object. The `IAgoraService` object persists as long as the server app keeps running.
+
+The SDK supports user IDs in both int format and string format. This page only shows user IDs in int format (the character set can only be digits). To learn more about user IDs in string format, see [Using User IDs in String Format](../develop/stringuid-c).
+
+```C++
```
Comment (yamasite) [inline shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:20]: Code highlighting looks off. Maybe something is wrong with the rendering.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+  }
+```
+
+### Step 2: Connect to the Agora RTC Channel
+
+After initializing the SDK, you can refer to the following steps to connect to the Agora RTC channel.
+
+1. Call `createRtcConnection` to create an `IRtcConnection` object to connect to the Agora RTC channel:
+
+    ```C++
```
Comment (yamasite) [inline shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:46]: Code highlighting looks off. Maybe something is wrong with the rendering.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+    agora::rtc::RtcConnectionConfiguration ccfg;
+    ccfg.autoSubscribeAudio = false;
+    ccfg.autoSubscribeVideo = false;
+    ccfg.clientRoleType = agora::rtc::CLIENT_ROLE_BROADCASTER;
+    agora::agora_refptr<agora::rtc::IRtcConnection> connection = service->createRtcConnection(ccfg);
+    ```
+
+2. Call `registerObserver` to listen to connection events. The `SampleConnectionObserver` class in the sample code inherits the `IRtcConnectionObserver` class and the `INetworkObserver` class:
+
+    ```C++
```
Comment (yamasite) [inline shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:57]: Code highlighting looks off. Maybe something is wrong with the rendering.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+    // Calls registerObserver to listen to connection events
+    auto connObserver = std::make_shared<SampleConnectionObserver>();
+    connection->registerObserver(connObserver.get());
+    ```
+
+3. Call `connect` to connect to an Agora RTC channel:
+
+    ```C++
```
Comment (yamasite) [inline shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:65]: Code highlighting looks off. Maybe something is wrong with the rendering.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx [hunk 7]
```diff
@@ -0,0 +1,540 @@
...
+        factory->createVideoEncodedImageSender();
+    if (!videoEncodedFrameSender) {
+    AG_LOG(ERROR, "Failed to create encoded video frame sender!");
+    return -1;
+    }
+    ```
+
+3. Create an `ILocalAudioTrack` object and an `ILocalVideoTrack` object, which correspond to the local audio track and local video track to be published to the Agora RTC channel:
+
+    ```c++
```
Comment (yamasite) [inline shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:135]: Code highlighting looks off. Maybe something is wrong with the rendering.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx [hunk 8]
```diff
@@ -0,0 +1,540 @@
...
+    AG_LOG(ERROR, "Failed to create video track!");
+    return -1;
+    }
+    ```
+
+### Step 2: Send media streams
+
+1. Use `publish` methods in the `ILocalUser` object to publish local audio and video tracks created in the previous step to the Agora RTC channel:
+
+    ```c++
```
Comment (yamasite) [inline shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:173]: Code highlighting in the following snippets looks off.

**Review by yamasite — CHANGES_REQUESTED** [review]

**Review by yamasite — CHANGES_REQUESTED** [review]

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx (hunk 3 above)
Comment (dengyaxi) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:20]: @FayHaifei Change C++ to `cpp` and see if this warning disappears when running `yarn start`
image

File: shared/cloud-gateway/overview/_product-overview.mdx (hunk 1 above)
Comment (FayHaifei) [reply shared/cloud-gateway/overview/_product-overview.mdx:11]: Fixed.

File: shared/cloud-gateway/overview/_product-overview.mdx (hunk 2 above)
Comment (FayHaifei) [reply shared/cloud-gateway/overview/_product-overview.mdx:25]: Fixed.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx (hunk 3 above)
Comment (FayHaifei) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:20]: Fixed.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx (hunk 4 above)
Comment (FayHaifei) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:46]: Fixed.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx (hunk 5 above)
Comment (FayHaifei) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:57]: Fixed.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx (hunk 6 above)
Comment (FayHaifei) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:65]: Fixed.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx (hunk 7 above)
Comment (FayHaifei) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:135]: Fixed.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx (hunk 8 above)
Comment (FayHaifei) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:173]: Fixed.

File: shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx [hunk 9]
```diff
@@ -0,0 +1,540 @@
+import * as data from '@site/data/variables';
+
+<PlatformWrapper platform="linux">
+This page introduces how to use the Cloud Gateway to send media streams to the client and receive media streams from the client.
+
+## Prerequisites
+
+Before you begin, make sure you have downloaded the latest Cloud Gateway. See [Run the Sample Project](../get-started/integrate-c-sdk).
```
[the remark this answers is not in the document]
Comment (FayHaifei) [reply shared/cloud-gateway/develop/_send-receive-media-streams-c.mdx:8]: Fixed.

**Comment by yamasite** [comment]
https://docs.agora.io/en/server_gateway/server_gateway_video_mixing?platform=Linux

**Comment by yamasite** [comment]
@dengyaxi One article added for Cloud Gateway:

https://docs.agora.io/en/server_gateway/server_gateway_video_mixing?platform=Linux

**Pull request merged**